packages/@aws-cdk/aws-gamelift/lib/game-session-queue.ts

```diff
@@ -0,0 +1,471 @@
+  metricPlacementsSucceeded(props?: cloudwatch.MetricOptions): cloudwatch.Metric;
+
+  /**
+   * Game session placement requests that reached the queue's timeout limit without being fulfilled since the last report.
+   */
+  metricPlacementsTimedOut(props?: cloudwatch.MetricOptions): cloudwatch.Metric;
+}
+
+/**
+   * A full specification of an gameSessionQueue that can be used to import it fluently into the CDK application.
+   */
+export interface GameSessionQueueAttributes {
+  /**
+   * The ARN of the gameSessionQueue
+   *
+   * At least one of `gameSessionQueueArn` and `gameSessionQueueName` must be provided.
+   *
+   * @default derived from `gameSessionQueueName`.
+   */
+  readonly gameSessionQueueArn?: string;
+
+  /**
+   * The name of the gameSessionQueue
+   *
+   * At least one of `gameSessionQueueName` and `gameSessionQueueArn`  must be provided.
+   *
+   * @default derived from `gameSessionQueueArn`.
+   */
+  readonly gameSessionQueueName?: string;
+}
+
+/**
+   * Properties for a new Fleet gameSessionQueue
+   */
+export interface GameSessionQueueProps {
+  /**
+       * Name of this gameSessionQueue
+       */
+  readonly gameSessionQueueName: string;
+
+  /**
+   *  Information to be added to all events that are related to this game session queue.
+   *
+   * @default no customer event data
+   */
+  readonly customEventData?: string;
+
+  /**
+   * A list of locations where a queue is allowed to place new game sessions.
+   *
+   * Locations are specified in the form of AWS Region codes, such as `us-west-2`.
+   *
+   * For queues that have multi-location fleets, you can use a filter configuration allow placement with some, but not all of these locations.
+   *
+   * @default game sessions can be placed in any queue location
+   */
+  readonly allowedLocations?: string[];
+
+  /**
+   * An SNS topic is set up to receive game session placement notifications.
+   *
+   * @default no notification
+   *
+   * @see https://docs.aws.amazon.com/gamelift/latest/developerguide/queue-notification.html
+   */
+  readonly notificationTarget?: sns.ITopic;
+
+  /**
+   * A set of policies that act as a sliding cap on player latency.
+   * FleetIQ works to deliver low latency for most players in a game session.
+   * These policies ensure that no individual player can be placed into a game with unreasonably high latency.
+   * Use multiple policies to gradually relax latency requirements a step at a time.
+   * Multiple policies are applied based on their maximum allowed latency, starting with the lowest value.
+   *
+   * @default no player latency policy
+   */
+  readonly playerLatencyPolicies?: PlayerLatencyPolicy[];
+
+  /**
+   * Custom settings to use when prioritizing destinations and locations for game session placements.
+   * This configuration replaces the FleetIQ default prioritization process.
+   *
+   * Priority types that are not explicitly named will be automatically applied at the end of the prioritization process.
```

Comment:
I took this explanation from our official documentation. But this paragraph explain that a user can override default FleetIQ priority configuration but defining  this custom settings and then being able to influence how it will manage priorities depending on the the Type (prioritizing destinations) and Locations of a Fleet